app/src/main/java/com/mpieterse/stride/ui/layout/central/viewmodels/HabitViewerViewModel.kt

```
@@ -33,170 +52,313 @@ class HabitViewerViewModel @Inject constructor(
-                load(habitId)
-                // Emit event to notify home screen
-                eventBus.emit(AppEventBus.AppEvent.CheckInCompleted)
+    fun updateLocalName(newName: String) {
+        val id = _state.value.habitId
+        if (id.isEmpty()) return
+        nameOverrideService.updateHabitName(id, newName)
+        _state.update { it.copy(displayName = newName) }
+        viewModelScope.launch { eventBus.emit(AppEventBus.AppEvent.HabitNameChanged) }
+    }
+
+    fun updateHabitDetails(updated: HabitData) = viewModelScope.launch {
```

detekt (1.23.8)

[warning] The function updateHabitDetails appears to be too complex based on Cyclomatic Complexity (complexity: 37). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CyclomaticComplexMethod)